_inc/lib/core-api/class-wpcom-rest-field-controller.php

```diff
@@ -0,0 +1,308 @@
+			register_rest_field( $object_type, $this->field_name, array(
+				'get_callback' => array( $this, 'get_for_response' ),
+				'update_callback' => array( $this, 'update_from_request' ),
+				'schema' => $this->get_schema(),
+			) );
+		}
+	}
+
+	/**
+	 * Ensures the response matches the schema and request context.
+	 *
+	 * You shouldn't have to extend this method.
+	 */
+	function prepare_for_response( $value, $request ) {
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$schema = $this->get_schema();
+
+		$is_valid = rest_validate_value_from_schema( $value, $schema, $this->field_name );
+		if ( is_wp_error( $is_valid ) ) {
+			return $is_valid;
+		}
+
+		return $this->filter_response_by_context( $value, $schema, $context );
+	}
+
+	/**
+	 * Returns the schema's default value
+	 *
+	 * If there is no default, returns the type's falsey value.
+	 *
+	 * @param array $schema
+	 */
+	final public function default_value( $schema ) {
+		if ( isset( $schema['default'] ) ) {
+			return $schema['default'];
+		}
+
+		// If you have something more complicated, use $schema['default'];
+		switch ( isset( $schema['type'] ) ? $schema['type'] : 'null' ) {
+		case 'string' :
+			return '';
+		case 'integer' :
+		case 'number' :
+			return 0;
+		case 'object' :
+			return (object) array();
+		case 'array' :
+			return array();
+		case 'boolean' :
+			return false;
+		case 'null' :
+		default :
+			return null;
+		}
+	}
+
+	/**
+	 * The field's wrapped getter. Does permission checks and output preparation.
+	 *
+	 * This cannot be extended: implement `->get()` instead.
+	 *
+	 * @param mixed $object_data Probably an array. Whatever the endpoint returns.
+	 * @param string $field_name Should always match `->field_name`
+	 * @param WP_REST_Request
+	 * @param $object_type Should always match `->object_type`
+	 */
+	final public function get_for_response( $object_data, $field_name, $request, $object_type ) {
+		$permission_check = $this->get_permission_check( $object_data, $request );
+
+		if ( ! $permission_check || is_wp_error( $permission_check ) ) {
+			return $this->default_value( $this->get_schema() );
+		}
+
+		$value = $this->get( $object_data, $request );
+
+		return $this->prepare_for_response( $value, $request );
+	}
+
+	/**
+	 * The field's wrapped setter. Does permission checks.
+	 *
+	 * This cannot be extended: implement `->update()` instead.
+	 *
+	 * @param mixed $value The new value for the field.
+	 * @param mixed $object_data Probably a WordPress object (e.g., WP_Post)
+	 * @param string $field_name Should always match `->field_name`
+	 * @param WP_REST_Request
+	 * @param $object_type Should always match `->object_type`
+	 */
+	final public function update_from_request( $value, $object_data, $field_name, $request, $object_type ) {
+		$permission_check = $this->update_permission_check( $value, $object_data, $request );
+
+		if ( ! $permission_check ) {
+			return;
```

Comment:
After reading your comment, I thought the answer was "yes". But I looked more into the Core code this talks to, and I think it's an implementation error to return false here.

I added some warning in ac026b3e1. We can make it return an error instead of the warning, though, if we think an error is better.